gateway/app/main/routes.py

```diff
@@ -52,6 +54,76 @@ def get_connected_nodes():
     )
 
 
+@main.route("/choose-model-host", methods=["GET"])
+def choose_model_host():
+    """ Used to choose some grid node to host a model.
+        PS: Currently we perform this randomly.
+    """
+    if n_replica:
+        hosts = random.sample(list(grid_nodes.keys()), n_replica)
+    else:
+        hosts = random.sample(list(grid_nodes.keys()), 1)
+    hosts_info = [(host, grid_nodes[host]) for host in hosts]
+    return Response(json.dumps(hosts_info), status=200, mimetype="application/json")
+
+
+@main.route("/search-model", methods=["POST"])
+def search_model():
+    """ Search for a plain text model on the grid network. """
+    body = json.loads(request.data)
+
+    # Invalid body
+    if "model_id" not in body:
+        return Response("", status=400, mimetype="application/json")
+
+    match_nodes = []
+    for node in grid_nodes:
+        try:
+            response = requests.get(grid_nodes[node] + "/models/").content
+        except:
```

Comment:
What exception do you expect here? Set this explicitly

Reply:
Done! (6ace96c)